### ISSUE TITLE
Interlaced modes disables 5x crop

### ISSUE THREAD

**Issue by Gazpr0m** [MiSTer-devel/Genesis_MiSTer#198]
When starting Sonic 2 in 2p mode the game changes video to a interlaced signal.
That disables the 5x crop instantly.

The only way to get 5x enabled again is to re-select the config file so the core restarts.
Just resetting the core doesn't fix it. 
I tried with vsync_adjust=0 too with the same result.

Exact rom used:
Sonic The Hedgehog 2 (World) (Rev B)

**Comment by birdybro** [comment]
Strange. I cannot reproduce this bug. I load sonic 2, go into two player, 5x works there. I'm running in the regular 1080p60 video mode option. 

Try and do "Save Settings" in the OSD to see if overwriting your cfg file with a new one changes behavior.

**Comment by Gazpr0m** [comment]
Tried saving the config again but no luck.
Can you please try replacing your all your genesis config files with the ones attached to see if it makes a difference?
[config.zip](https://github.com/MiSTer-devel/Genesis_MiSTer/files/7834797/config.zip)

**Comment by asturur** [comment]
Hi, i reuse this issue to ask a question around cropping in general.
Since i switched from 1440p to my 1080p i find myself cropping all the time.
In general cores are setup ( this included ) to enable crop option only if scale mode is left at 0, so not forced integer mode.
That would be fine, because cropping is then set up to create an integer scale mode with 1080p.

Now the issue is that for all the cores that do not support cropping nor scale mode selection, integer scaling is far superior than full screen scaling, and the setting is inherited from mister.ini

Is there a reason why integer scale in the settings.ini disable support for cropping?

reference: https://github.com/MiSTer-devel/Genesis_MiSTer/blob/master/Genesis.sv#L221

I was about to test and verify and open a PR on all cores.
But i figured out talking first and getting context is always better.

Thanks.

**Comment by sorgelig** [comment]
Yes, integer scaling in INI is incompatible with cropping due to the way it works.
In theory core may provide some cropping expecting integer scaling from ini to be used. Scaler itself doesn't support cropping. Each core should provide a carefully implemented crop suitable for the core. Console cores almost all have cropping support already. Some computer cores with large borders also have cropping.
For arcades - sorry they are too many to work with it.

In general Integer scaling is crap. Having video with borders all around is doubtful pleasure. Today with video filter with shadow mask and scanlines (scandoubler scanlines with special filter on top) can minimize effect of non-integer scaling almost to the point as it was on CRT.

**Comment by asturur** [comment]
So if i recompile the core removing the constrain, video_freak module should not work anyway with integer scaling forced by ini.
right? ( i m sure i ll try too see it straight ).

So today we have:
- force scandoubler from ini
- scanlines/scandoubling and hq2x from video menu in all the cores
- integer scaling with options from ini
- integer scaling with some options in cores
- cropping that works only with some combination of options
- video filters that work great always without scanlines, often with integer scaling + shadowmasks.

I wonder if some option became really superflous and could be removed for clarity.



> Yes, integer scaling in INI is incompatible with cropping due to the way it works. In theory core may provide some cropping expecting integer scaling from ini to be used. Scaler itself doesn't support cropping. Each core should provide a carefully implemented crop suitable for the core. Console cores almost all have cropping support already. Some computer cores with large borders also have cropping. For arcades - sorry they are too many to work with it.

> 
> In general Integer scaling is crap. Having video with borders all around is doubtful pleasure. Today with video filter with shadow mask and scanlines (scandoubler scanlines with special filter on top) can minimize effect of non-integer scaling almost to the point as it was on CRT.

Well, yes integer scaling alone is a bit off, but with cropping, when you don't have to crop too much, is great.

Video filters are cool too, but sometimes i love pixels.
Do we know how many ms takes a full blown filter setup? ( 2 pass v + h, gamma and shadowmask )

**Comment by sorgelig** [comment]
> Do we know how many ms takes a full blown filter setup? ( 2 pass v + h, gamma and shadowmask )

Couple of pixels. All that processing doesn't use additional frame buffer. It's done on the fly without latency.

**Comment by asturur** [comment]
when you say couple of pixels you mean the window of the horizontal filter taps? How does the vertical filter then work?

I m really curious to understand, do we have any technical docs on the filters that do not require reverse engineering the verilog code?

**Comment by birdybro** [comment]
Horizontal filter is personally easily explained like this...

Stick your finger straight into the air and keep it straight up, now move your hand left to right. That is filtering the video along the horizontal axis.

![image](https://user-images.githubusercontent.com/16388068/151709018-4171df12-0931-4820-8b3e-8c5647f90501.png)

Now do the same with your finger pointed to the left/right and move your hand up and down, that is filtering video along the vertical axis.

![image](https://user-images.githubusercontent.com/16388068/151709028-5a232268-0025-47f9-a951-67c6c7588c56.png)

**Comment by asturur** [comment]
@birdybro  thanks for the explanation. I meant something more technical.

The horizontal filter can filter based on the information of the current line, so i can understand it can act without a buffer. If it has, i don't know, 5 taps, will read the current pixel, the previous 2 and will wait the next 2, and will do its calculation.
So when @sorgelig says, the time used is a couple of pixels, i think i can grasp what he is saying.

But when you go for vertical filtering, you have to wait for a full lines to render, and that is fine too, i still very fast, 2-4 lines on a 240 output is a 1/60th of a frame, nothing.

I m just curious to learn. i also wanted to build a simulator but i don't know how filters behave and pulling it out from verilog is prone to errors if you are not expert.

Also, seems from the effects i see in filter, the filter act after the scaler has scaled.
Does it mean it has to scan the full resolution ouput?
Is there i can check in the wiki?

**Comment by sorgelig** [comment]
Yes, vertical filter needs up to 3 lines to process if i remember right. 
There is no such very technical info in MiSTer Wiki. You need to search about polyphase scaling filter in internet.

**Comment by birdybro** [comment]
@asturur 

https://github.com/rikard-softgear/PolyphasePreviewer

Here's a simulator of the polyphase filtering coded in c# from the previous iteration, but it is not updated for 64-phase yet. This might be a decent starting place if you are familiar with c#. :)

**Comment by asturur** [comment]
thank you! i should be able to read it!
